var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -76,11 +76,38 @@ def install(self, spec, prefix):
 
         if '+blas' in spec or '+lapack' in spec:
             with open('site.cfg', 'w') as f:
-                f.write('[DEFAULT]\n')
-                f.write('libraries=%s\n'    % ','.join(lapackblas.names))
+                if '^mkl' in spec:
+                    # Even though the intel-parallel-studio package provides
+                    # mkl that package will not work here with a non-Intel
+                    # compiler due to header collision.
+                    f.write('[mkl]\n')
+                    f.write('libraries=mkl_rt\n')
```

Comment:
i am not in favour of this change. `join(lapackblas.names)` depends on whole lot of variants (32/64 bits, threaded/serial, and alike). Once you use `mkl_rt` here it is now up to environment variable. In other words you introduce a way to use different BLAS/LAPACK in DAG. 

What was wrong exactly with `'join(lapackblas.names)`? At the end of the day we give the complete library lists there and it is bound to work. At least it does in q number of other packages without a need to use `mkl_rt` anywhere.